unlock-app/src/reducers/errorsReducer.js

```diff
@@ -10,16 +10,30 @@ const errorsReducer = (state = initialState, action) => {
   }
 
   if (action.type === SET_ERROR) {
-    if (state.includes(action.error)) return state
-    return [...state, action.error]
+    if (state.map(error => error.name).includes(action.error)) return state
```

Comment:
```js
if (state.find(error => action.error === error.name && action.data !== error.data)) return state
```

also need to test for the equality of data

Reply:
Nope. I actually think this is fine. The data is not defining what the error is.

Reply:
ok. this is a nit, but still more efficient to use:

```js
if (state.find(error => action.error === error.name)) return state
```

However, there is an example of a situation where the data defines what the error is:

```
user navigates to paywall, but is on rinkeby
"wrong network" error is dispatched, with data being rinkeby
user quickly tries to change to mainnet, but accidentally clicks Ropsten
"wrong network" error is dispatched, with data being ropsten
error message does not update
```

Technically, it is still the same kind of error, but if we don't update the state, the change in the data is lost in the UI's display of the error.

One possible solution is to make separate errors for each data possibility, but I think in this case it may be difficult, as a user could use their own custom ethereum network, and we would not have an error for it.

Reply:
Well you can;t be on 2 networks at once, can you? So in practice the metadata helps us make a better tailored message to the user but does not define the error.

Reply:
When swicting the network the app  will reload... so state is reset no matter what.

Reply:
This is just an example. If we don't save the error, the change in data is lost to the UI. I'll approve this, but with a strong recommendation of changing this since it will be a difficult thing to debug if it ever happens in production.